src/sedpack/io/dataset_filler.py

```diff
@@ -153,7 +152,8 @@ def write_example(
 
         # New metadata
         previous_metadata = current_progress.shard.shard_info.custom_metadata
-        metadata_changed: bool = (custom_metadata and previous_metadata
+        metadata_changed: bool = (bool(custom_metadata)
+                                  and bool(previous_metadata)
```

Comment:
Might be more readable using `all()`
Caveat: IIRC `all()` evaluates all expressions first whereas boolean `and` applies lazy evaluation and stops at the first item evaluated to `false`.